var/spack/repos/builtin/packages/py-prometheus-client/package.py

```diff
@@ -21,9 +22,10 @@ class PyPrometheusClient(PythonPackage):
     depends_on("py-setuptools", type="build")
     # Notice: prometheus_client/twisted/_exposition.py imports 'twisted.web.wsgi'
     # which was not ported to Python 3 until twisted 16.0.0
-    depends_on("py-twisted", type=("build", "run"), when="+twisted")
-    depends_on("py-twisted@16:", type=("build", "run"), when="@0.12.0: +twisted ^python@3:")
-    depends_on("python@2.7:2,3.4:", type=("build", "run"), when="@0.12.0:")
+    depends_on("py-twisted", when="+twisted", type=("build", "run"))
+    depends_on("py-twisted@16:", when="@0.12.0: +twisted ^python@3:", type=("build", "run"))
+    depends_on("python@2.7:2,3.4:", when="@0.12.0", type=("build", "run"))
+    depends_on("python@3.6:", when="@0.14.1:", type=("build", "run"))
```

Comment:
Technically you could remove both these lines since Spack only supports Python 3.7+

Reply:
Let's leave this for a follow up.